lib/fs.js

```diff
@@ -453,40 +454,86 @@ function closeSync(fd) {
   handleErrorFromBinding(ctx);
 }
 
-function open(path, flags, mode, callback) {
+// This function has two possible signatures:
+// fs.open(path[, flag[, mode]], callback)
+// fs.open(path[, options], callback)
+function open(path, flag, mode, callback) {
+  let options;
+  let parents;
   path = getValidatedPath(path);
   if (arguments.length < 3) {
-    callback = flags;
-    flags = 'r';
+    callback = flag;
+    flag = 'r';
     mode = 0o666;
   } else if (typeof mode === 'function') {
     callback = mode;
     mode = 0o666;
-  } else {
-    mode = parseFileMode(mode, 'mode', 0o666);
   }
-  const flagsNumber = stringToFlags(flags);
+  // Handle case where 2nd argument is options-bag
+  if (typeof flag === 'object' && flag) {
+    options = flag;
+    flag = options.flag ?? 'r';
+    mode = options.mode ?? 0o666;
+    parents = options.parents;
+  }
+  flag = stringToFlags(flag);
```

Comment:
I think it should be `flags` everywhere, not `flag`.

Reply:
k, added shim converting `writeFile[Sync]:flag` to `open[Sync]:flags` in commit 7b5b3e14a8d8a5c9f3fa18fed1cb42d004ef4692

```diff
 async function writeFile(path, data, options) {
   options = getOptions(options, { encoding: 'utf8', mode: 0o666, flag: 'w' });
-  const flag = options.flag || 'w';
+  // Don't make changes directly on options object
+  options = copyObject(options);
+  // flag to flags shim:
+  // Resolve inconsistency between writeFile[Sync]:flag to open[Sync]:flags
+  options.flags = options.flag ?? 'w';
+  delete options.flag;

...
-  const fd = await open(path, flag, options.mode);
+  const fd = await open(path, options);
```